dev/Kozani/KozaniManager/main.cpp

```diff
@@ -33,20 +87,93 @@ struct __declspec(uuid(PR_KOZANIMANAGER_CLSID_STRING)) KozaniManagerImpl WrlFina
         PCWSTR additionalSettingsFilePath,
         ::IInspectable* activatedEventArgs,
         IKozaniStatusCallback* statusCallback,
-        DWORD associatedLocalProcessId) try
+        DWORD associatedLocalProcessId) noexcept try
     {
-        // Serialize the message
-        const std::int64_t cookie{};
-        std::string messageBytes{ ::Microsoft::Kozani::Activation::ActivateApp(cookie, appUserModelId, activatedEventArgs) };
+        auto activationKindLocal{ static_cast<winrt::Windows::ApplicationModel::Activation::ActivationKind>(activationKind) };
+        RETURN_HR_IF_MSG(HRESULT_FROM_WIN32(ERROR_NOT_SUPPORTED), 
+            !IsActivationKindSupported(activationKindLocal),
+            "Unsupported activation kind: %d", activationKind);
```

Comment:
You will need to look up the error code to know it means unsupported. Always easier to have a human readable error message than a simplified message that needs extra decoding, unless log size is a concern.

Reply:
Beware "human readable" as you trend into localization territory...and that's not something we want to incur here for WIL logging. Plus "pretty english prose WIL log messages" add up quickly leading to PE file bloat with negative product implications.

And since this is our plumbing - our process, not an inproc API in some dev's app's process - the _real_ support answer is EventLog logging. Don't try to expand WIL to fill that niche.

Standard coding guidelines is to avoid unnecessary WIL LOG* message strings. Data in LOG*MSG() is awesome as it can provide critical context what a codepath is operating on but take the minimalist approach as the source will always be available and, commonly, necessary.

Reply:
Defined KOZANI_E_UNSUPPORTED_ACTIVATION_KIND and shortened the message: "Activation kind: %d"